File: cpp/bc1914.cpp
#include <iostream>

using namespace std;

int main()
{
	int nCa, fPe, sPe;
	string fP, fVal, sP, sVal, res="";
	
	cin >> nCa;
	
	for(int i = 0; i < nCa; i++)
	{
		cin >> fP >> fVal >> sP >> sVal;
		cin >> fPe >> sPe;
		
		((fPe + sPe) % 2) == 0? res = "PAR" : res = "IMPAR";
		
		if(res == fVal) cout << fP << endl;
		else cout << sP << endl;
	}
}

File: python/bc1914.py
nCa = int(input())

for c in range(0, nCa):
    line = input().split(" ")

    fP = line[0]
    fCh = line[1]
    sP = line[2]
    sCh = line[3]

    cVal = input().split(" ")

    fPe = int(cVal[0])
    sPe = int(cVal[1])

    res = "PAR" if ((fPe + sPe) % 2) == 0 else "IMPAR"

    print(f"{fP}" if res == fCh else f"{sP}")
